crates/handlers/src/compat/login.rs

```diff
@@ -506,7 +514,13 @@ async fn token_login(
             browser_session.id = %browser_session_id,
             "Attempt to exchange login token but browser session is not active"
         );
-        return Err(RouteError::InvalidLoginToken);
```

Comment:
I think it would be more readable to check if the user is locked earlier

```rust
if browser_session.user.locked_at.is_some() {
  return Err(RouteError::UserLocked);
}

if !browser_session.active() || !browser_session.user.is_valid() {
  // ...
}
```

It changes a little bit what error we show in what condition, but it's an edge-case anyway, as usually we would show a user comprehensible error earlier in the UI

Reply:
On second thought, I've reverted this with 05827d1e (the commit message explains why).  The logic is more precise this time, though.

Reply:
Alright, works for me!